synapse/models/inet.py

```diff
@@ -3310,6 +3478,269 @@ def getModelDefs(self):
+                            'doc': 'The name of the group on this platform.'}),
+
+                        ('profile', ('ps:contact', {}), {
+                            'doc': 'Current detailed contact information for this group.'}),
+                    )),
+
+                    ('inet:service:group:member', {}, (
+
+                        ('account', ('inet:service:account', {}), {
+                            'doc': 'The account that is a member of the group.'}),
+
+                        ('group', ('inet:service:group', {}), {
+                            'doc': 'The group that the account is a member of.'}),
+
+                        ('period', ('ival', {}), {
+                            'doc': 'The time period when the account was a member of the group.'}),
+                    )),
+
+                    ('inet:service:permission:type:taxonomy', {}, ()),
+
+                    ('inet:service:permission', {}, (
+
+                        ('name', ('str', {'onespace': True, 'lower': True}), {
+                            'doc': 'The name of the permission.'}),
+
+                        ('type', ('inet:service:permission:type:taxonomy', {}), {
+                            'doc': 'The type taxonomy for the permissions.'}),
+
+                    )),
+
+                    ('inet:service:rule', {}, (
+
+                        ('permission', ('inet:service:permission', {}), {
+                            'doc': 'The permission which is granted.'}),
+
+                        ('denied', ('bool', {}), {
+                            'doc': 'Set to (true) to denote that the rule is an explicit deny.'}),
+
+                        ('object', ('ndef', {'interface': 'inet:service:object'}), {
+                            'doc': 'The object that the permission controls access to.'}),
+
+                        ('grantee', ('ndef', {'forms': ('inet:service:account', 'inet:service:group')}), {
+                            'doc': 'The user or role which is granted the permission.'}),
+                    )),
+
+                    ('inet:service:session', {}, (
+
+                        ('id', ('str', {'strip': True}), {
+                            'doc': 'The service specific session id.'}),
+
+                        ('creator', ('inet:service:account', {}), {
+                            'doc': 'The account which authenticated to create the session.'}),
+
+                        ('period', ('ival', {}), {
+                            'doc': 'The period where the session was valid.'}),
+                    )),
+
+                    ('inet:service:login', {}, (
+
+                        ('method', ('inet:service:login:method:taxonomy', {}), {
+                            'doc': 'The type of authentication used for the login. For example "password" or "multifactor.sms".'}),
+
+                        # TODO ndef based auth proto details
+                    )),
+
+                    ('inet:service:message', {}, (
```

Comment:
Just noticed, do we need an inet:service:message:url to note the URL where the message is posted/viewable, if any?

Reply:
Yep! I'll add.